syft/federated/plan.py

```diff
@@ -254,12 +254,21 @@ def replace_worker_ids(self, from_worker_id: Union[str, int], to_worker_id: Unio
             from_worker_id: Id of the the replaced worker.
             to_worker_id: Id of the new worker.
         """
-        for from_id, to_id in [
-            (from_worker_id, to_worker_id),
-            (from_worker_id.encode(), to_worker_id.encode()),
-        ]:
+
+        id_pairs = [(from_worker_id, to_worker_id)]
+        if type(from_worker_id) == str:
+            to_worker_id_encoded = (
+                to_worker_id.encode() if type(to_worker_id) == str else to_worker_id
```

Comment:
Can you cover all 4 possibilities when dealing with strings?

from_worker, to_worker
from_worker, to_worker.encode()
from_worker.encode(), to_worker
from_worker.encode(), to_worker.encode()

Reply:
We don't need all four possibilities. The outdated code in dev contained the four possibilities. Then the current dev code changed it to (from_worker, to_worker) and (from_worker.encode(), to_worker.encode()) but without taking care of the integers that don't have a encode() function.

As we are replacing, it doesn't make sense to replace from_worker twice. Therefore only two outcomes, not four.

Reply:
Hmm, I would say we need the 4 possibilities (as in train-config) to cover all possible cases, don't we?

My question then is: can we guarantee that with only these two cases all the other cases will work?

Reply:
Let's think of an example doing string replacements:
"The quick brown b'brown' fox jumps over the lazy b'brown' dog."
If we want to replace brown by black, we would call the replace method with from_worker=brown and to_worker=black.
First call would replace all brown instances with black:
"The quick black b'brown' fox jumps over the lazy b'brown' dog."
The second call would replace all b'brown' instances with b'black':
"The quick black b'black' fox jumps over the lazy b'black' dog."
And we are done.